src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -197,17 +210,26 @@ private bool IsServerValidForCluster(ClusterType clusterType, ClusterConnectionM
                     return serverType == ServerType.ShardRouter;
 
                 case ClusterType.Unknown:
-                    switch (connectionMode)
+#pragma warning disable CS0618
+                    if (clusterSettings.ConnectionModeSwitch != ConnectionModeSwitch.UseDirectConnection)
                     {
-                        case ClusterConnectionMode.Automatic:
-                            if (serverType == ServerType.Standalone)
-                            {
-                                return Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv; // Standalone is only valid in MultiServerCluster when using MongoDBPlusSrv scheme
-                            }
-                            return serverType.IsReplicaSetMember() || serverType == ServerType.ShardRouter;
+                        switch (clusterSettings.ConnectionMode)
+                        {
+                            case ClusterConnectionMode.Automatic:
+#pragma warning restore CS0618
+                                if (serverType == ServerType.Standalone)
+                                {
+                                    return _servers.Count == 1 || Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv; // Standalone is only valid in MultiServerCluster when using MongoDBPlusSrv scheme
+                                }
+                                return serverType.IsReplicaSetMember() || serverType == ServerType.ShardRouter;
 
-                        default:
-                            throw new MongoInternalException("Unexpected connection mode.");
+                            default:
+                                throw new MongoInternalException("Unexpected connection mode.");
+                        }
+                    }
+                    else
+                    {
+                        return true;
```

Comment:
Replace 213-233 with:

    #pragma warning disable CS0618
                        if (clusterSettings.ConnectionModeSwitch == ConnectionModeSwitch.UseDirectConnection)
                        {
                            return true;
                        }
                        else
                        {
                            switch (clusterSettings.ConnectionMode)
                            {
                                case ClusterConnectionMode.Automatic:
                                    if (serverType == ServerType.Standalone)
                                    {
                                        return _servers.Count == 1 || Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv; // Standalone is only valid in MultiServerCluster when using MongoDBPlusSrv scheme
                                    }
                                    return serverType.IsReplicaSetMember() || serverType == ServerType.ShardRouter;

                                default:
                                    throw new MongoInternalException("Unexpected connection mode.");
                            }
                        }
    #pragma warning restore CS0618